nvptx-run.c

```diff
@@ -235,6 +279,27 @@ This program has absolutely no warranty.\n",
   r = cuCtxCreate (&ctx, 0, dev);
   fatal_unless_success (r, "cuCtxCreate failed");
 
+  size_t mem;
+  if (!stack_size || verbose)
+    {
+      r = cuDeviceTotalMem (&mem, dev);
+      fatal_unless_success (r, "could not get available memory");
+      report_val (stderr, "Total device memory", mem);
+    }
+
+  size_t free_mem;
+  size_t dummy;
+  if (verbose)
+    {
+      /* Set stack size limit to 0 to get more accurate free_mem.  */
+      r = cuCtxSetLimit(CU_LIMIT_STACK_SIZE, 0);
```

Comment:
From `cuCtxSetLimit`: <https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__CTX.html#group__CUDA__CTX_1g0651954dfb9788173e60a9af7201e65a> I can't easily tell the rationale here.

So, should we add more commentary for this, or point to an external URL if that makes sense?